CMakeLists.txt

```
@@ -330,6 +330,8 @@ endif()
 # This will only run if python3 is available.
 #-------------------------------------------------------------------------------
 
+option(IREE_WARN_ON_MISSING_SUBMODULES "Only warn if submodules have not been initialized." OFF)
```

The name of this is a bit confusing to me because it suggests it turns a warning on, but really it turns an error off. What about instead making this an "enum" for the behavior

```cmake
set(IREE_MISSING_SUBMODULES CACHE STRING "ERROR"
      "Behavior when missing submodules are detected: OFF, WARN, or ERROR" )
```

---

Agreed it's confusing, though an on/off option is much simpler than an enum. I'd also rather remove the TF submodule or rework how deps are included. This sort of "sometimes warn about a state based on the specific configuration" CMake/git hacks (and much of the configuration in this room CMakeLists.txt) should be reviewed for more cleanup soon.

---

How about just ERROR and off then

```suggestion
option(IREE_ERROR_ON_MISSING_SUBMODULES "Error if submodules have not been initialized." ON)
```

---

Thanks for the suggestion :) I agree and followed your proposal @GMNGeoffrey.

> I'd also rather remove the TF submodule or rework how deps are included.

I absolutely agree @ScottTodd! That is on the list next, but much easier to "check" what those changes to as an `add_directory` simply fails on an uninitialized submodule/when no `CMakeLists.txt` is present.